elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/engine/SQLQueryEngine.java

```diff
@@ -175,10 +184,140 @@ public SQLQueryEngine(EntityManager entityManager, EntityDictionary dictionary)
 
         MutableInt counter = new MutableInt(0);
 
-        return results.stream()
+        // mapping: relationship field name -> join ID's
+        Map<String, List<Object>> hydrationIdsByRelationship = new HashMap<>();
+        List<Object> queryResults = results.stream()
                 .map((result) -> { return result instanceof Object[] ? (Object []) result : new Object[] { result }; })
-                .map((result) -> coerceObjectToEntity(query, result, counter))
+                .map((result) -> coerceObjectToEntity(query, result, counter, hydrationIdsByRelationship))
                 .collect(Collectors.toList());
+
+        // hydrate each relationship
+        for (Map.Entry<String, List<Object>> entry : hydrationIdsByRelationship.entrySet()) {
+            String joinField = entry.getKey();
+            List<Object> joinFieldIds = entry.getValue();
+
+            List<Object> relationshipValues = getRelationshipValues(query.getEntityClass(), joinField, joinFieldIds);
+
+            for (int i = 0; i < relationshipValues.size(); i++) {
+                Object entityInstance = queryResults.get(i);
+                dictionary.setValue(entityInstance, joinField, relationshipValues.get(i));
+            }
+        }
```

Comment:
If we would like to generalize it, we could move this part into a parent class and have `SQLQueryEngine` implement  `getRelationshipValues` below